data/colors/vars/component_dark.ts

```diff
@@ -186,5 +186,20 @@ export default {
       bg: get('scale.blue.5'),
       disabledBg: get('scale.gray.5'),
     }
-  }
+  },
+
+  segmentedControl: {
+    bg: get('scale.gray.8'),
+    hoverBg: get('scale.gray.6'),
+
+    active: {
+      bg: get('scale.gray.7'),
+      border: get('scale.gray.6'),
+    },
+
+    pressed: {
+      bg: get('scale.gray.6'),
+      border: get('scale.gray.4'),
```

Comment:
Based on https://github.com/github/primer/issues/762#issuecomment-1138961381, `gray.4` might not be enough when the `SegmentedControl` is being used on a darker background like for the `CommentBox`:

<img width="180" alt="image" src="https://user-images.githubusercontent.com/378023/171622239-5b7a7816-620d-4b27-b37e-f6be6ff511cd.png">